hs_core/views/__init__.py

```diff
@@ -382,6 +382,82 @@ def unshare_resource_with_user(request, shortkey, user_id, *args, **kwargs):
 
     return HttpResponseRedirect(request.META['HTTP_REFERER'])
 
+
+def resource_labeling_action(request, shortkey, *args, **kwargs):
```

Comment:
This view belongs in the `hs_labels` app. The `hs_core` app is already enormous.